util/datetime.go

```diff
@@ -6,20 +6,32 @@ import (
 )
 
 const (
-	shortFormat string = "2006 Jan 02, 15:04"
+	shortFormat string = "2006 Jan 02, 15:04 UTC"
 )
 
 // ParseDate parses our common date/time strings
 // and returns a time.Time object
 func ParseDate(dateString string) time.Time {
 	// our standard format
 	template := "2006-01-02T15:04:05.000Z"
+	alternativeTemplate := "2006-01-02T15:04:05.000-07:00"
 
 	// normalizing the number of decimal places to 3
-	re := regexp.MustCompile("\\.[0-9]+Z$")
-	dateString = re.ReplaceAllLiteralString(dateString, ".000Z")
+	// and discarding sub-second detail along the way
+	re := regexp.MustCompile("\\.[0-9]+")
+	dateString = re.ReplaceAllLiteralString(dateString, ".000")
 
-	t, _ := time.Parse(template, dateString)
+	// try parsing with several formats
+	t, err := time.Parse(template, dateString)
+	if err == nil {
+		return t
```

Comment:
I kind of like the fact that the functions can be used like `created := util.ShortDate(util.ParseDate(cluster.CreateDate))`. The worst thing that can happen here is that a bad date ( 0000-01-01 ...) gets displayed.